Makefile_i8_PaStiX

```diff
@@ -0,0 +1,131 @@
+# See our wiki for getting the CalculiX dependencies:
+# https://github.com/precice/calculix-adapter/wiki/Installation-instructions-for-CalculiX
+# Set the following variables before building:
+# Path to original CalculiX source (e.g. $(HOME)/ccx_2.16/src )
+CCX             = $(HOME)/CalculiX/ccx_2.17/src
+
+### Change these if you built SPOOLES, ARPACK, or yaml-cpp from source ###
+# SPOOLES include flags (e.g. -I$(HOME)/SPOOLES.2.2 )
+SPOOLES_INCLUDE   = -I/usr/include/spooles/
+# SPOOLES library flags (e.g. $(HOME)/SPOOLES.2.2/spooles.a)
+SPOOLES_LIBS      = -lspooles
+#
+# ARPACK include flags (e.g. -I$(HOME)/ARPACK)
+ARPACK_INCLUDE    =
+# ARPACK library flags (e.g. $(HOME)/ARPACK/libarpack_INTEL.a)
+ARPACK_LIBS       = $(HOME)/ARPACK/libarpack_INTEL.a 
+#ARPACK_LIBS       = -larpack -llapack -lblas
+#
+# yaml-cpp include flags (e.g. -I$(HOME)/yaml-cpp/include)
+YAML_INCLUDE      = -I/usr/include/
+# yaml-cpp library flags (e.g. -L$(HOME)/yaml-cpp/build -lyaml-cpp)
+YAML_LIBS         = -lyaml-cpp
+
+# Get the CFLAGS and LIBS from pkg-config (preCICE version >= 1.4.0).
+# If pkg-config cannot find the libprecice.pc meta-file, you may need to set the
+# path where this is stored into PKG_CONFIG_PATH when building the adapter.
+PKGCONF_CFLAGS  = $(shell pkg-config --cflags libprecice)
+PKGCONF_LIBS    = $(shell pkg-config --libs libprecice)
+
+# Specify where to store the generated .o files
+OBJDIR = bin
+
+# To get the correct path, export the required folders for pkg-config. Example: export PKG_CONFIG_PATH=~/PaStiX/pastix_i8/lib/pkgconfig/:~/PaStiX/hwloc_i8/lib/pkgconfig/:~/PaStiX/parsec_i8/lib/pkgconfig/
+
+PASTIX_INCLUDE = $(shell pkg-config --cflags pastix)
+PASTIX_LIBS = $(shell pkg-config --libs pastix hwloc parsec lapacke) $(HOME)/OpenBLAS_i8/libopenblas.a \
+$(HOME)/PaStiX/pastix_i8/lib/libpastix_kernels.a \
+   /usr/lib/x86_64-linux-gnu/libcublas_static.a \
+  /usr/lib/x86_64-linux-gnu/libculibos.a /usr/lib/x86_64-linux-gnu/libcublasLt_static.a \
+  $(HOME)/PaStiX/scotch_i8/lib/libscotch.a $(HOME)/PaStiX/scotch_i8/lib/libscotcherrexit.a 
```

Comment:
I understand that you are specifying all the dependencies of pastix here, because you are using pastix as a static library. This should be fine, but I feel it would be simpler to use pastix as a shared library, so that we don't need to specify all these. Is pastix available as a shared library? If not, can we get all these directly from the pkg-config of pastix?

Reply:
I already use the pkg-config for pastix, but it doesn't contain the `libpastix_kernels.a`, surprisingly.
As for shared PaStiX... Since we use a specific version of PaStiX designed for CalculiX I wouldn't trust the possibilities, since CalculiX gives very specific instructions.
However the CMakeFile of pastix does contain a SHARED mode, so I can give it a try.

Reply:
If this is indeed a special version, then let's keep it static to avoid any accidental linking to the wrong one.